### PR TITLE
Fix nlohman::json array list initializer broken on GCC

### PR DESCRIPTION
**Pull request by elpaso**
the issue:

`json j { something_returning_json_array_of_values() };`

produces a `json::array( json::array( values ) ) )`

instead of `json::array( values )`

by using the assignment all is fine:

`json j = something_returning_json_array_of_values();`

produces a `json::array( values )` as expected

### CODE REVIEW

**Comment by elpaso** [comment]
@m-kuhn I think you've already seen this this bug recently.

File: src/core/qgsjsonutils.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ json QgsJsonExporter::exportFeaturesToJsonObject( const QgsFeatureList &features
   const auto constFeatures = features;
   for ( const QgsFeature &feature : constFeatures )
   {
-    data["features"].push_back( exportFeatureToJsonObject( feature ) );
+    data[ "features" ].push_back( exportFeatureToJsonObject( feature ) );
```
Comment (m-kuhn) [inline src/core/qgsjsonutils.cpp]: Since this is a cosmetic one, can you use QStringLiteral on "features" and drop 238 in favor of `qgis::as_const`? ;)

**Comment by m-kuhn** [comment]
@signedav you looked into it in more detail, what do you think?

File: src/core/qgsjsonutils.cpp (hunk 1 above)
Comment (elpaso) [reply src/core/qgsjsonutils.cpp]: no literal here, it's a std::strings, ok for the as_const

**Comment by signedav** [comment]
I confirm what @elpaso writes regarding the json.

Like the observation I made during debugging and wrote down here https://github.com/qgis/QGIS/pull/30758:
> Apparently `{ QgsJsonUtils::jsonFromVariant( value ) }` returns an array containing one element and that's an object while `= QgsJsonUtils::jsonFromVariant( value )` contains the object itself not nested in an array.

**Comment by elpaso** [comment]
@signedav just to be sure, which compiler are you using? Because I was not able to reproduce this on clang, but only on GCC.

**Comment by signedav** [comment]
GCC as well.

**Comment by elpaso** [comment]
no need to backport

**Pull request merged**